src/views/MemoDetail.vue

```diff
@@ -0,0 +1,74 @@
+<template>
+  <div class="backdrop" @click.self='closeDetail'>
+    <div class="modal">
+      <h2>Memo Detail</h2>
+      <label>
+        <input placeholder="Title" v-model='memo.title'>
+      </label>
+      <label>
+        <textarea placeholder="Content" v-model='memo.content'></textarea>
+      </label>
+      <button @click='editMemo'>Edit</button>
+      <button @click='deleteMemo'>Delete</button>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'MemoDetail',
+  data () {
+    const data = {
+      memos: [],
+      memo: {
+        id: '',
+        title: '',
+        content: ''
+      }
+    }
+    return data
+  },
+  mounted () {
+    if (localStorage.getItem('memos-vuejs')) {
+      this.memos = JSON.parse(localStorage.getItem('memos-vuejs'))
+      this.memo = this.memos.find((memo) => memo.id == this.$route.params.id)
+    }
+  },
+  beforeRouteEnter(to, from, next) {
+    const memos = JSON.parse(localStorage.getItem('memos-vuejs'))
+    if (!memos) next({path: '/:catchAll(.*)'})
+    const id = memos.find((m) => m.id == to.params.id)
+    if (!id) next({path: '/:catchAll(.*)'})
```

Comment:
`Array.prototype.find()` を呼び出しているところは、memo が見つかった場合は memo そのものが返ってくるはずなので、以下のようにする必要がありそうです。

```suggestion
    const memo = memos.find((m) => m.id == to.params.id)
    if (!memo) next({path: '/:catchAll(.*)'})
```

Reply:
確かにメモが返ってきていました💦修正しました！